### DISCUSSION TITLE
Multiple menu locations in a future version?

### DISCUSSION THREAD

**Discussion by PortMacOnline** [holger1411/understrap#1332] [Ideas]
Please delete anything that does not apply for you!

## Issue is about..
  - [X] a Feature Request
  - [X] a Suggestion

## Please tell us about your environment:

Not relevant -- suggestion/feature request applies to basic functions/features. 

### Other Comments

Having only one menu location in 2021 is beyond belief -- please implement multiple menu locations in a future version. Other themes have them, so I know it can be done. 

Other possible locations:
above the main logo/menu location -- possible usage - membership login or shopping cart display
below the main logo/menu location -- possible usage - supplementary menu, as opposed to sub-menus
above the footer widget location -- currently used in other themes for a Legal Pages Menu -- Privacy Page, Terms Page, etc
below the footer widget location -- possible usage -- as alternative to above the footer widget menu location.

**Comment by danemorgan** [comment]
WordPress has a [philosophy](https://wordpress.org/about/philosophy/) about this. Supply the code 80% will use and leave the code only 20% will use to custom application. Understrap tries to adhere to the overall WordPress Philosophy.

At least 80% of all users will use that one menu location. I am not sure the same case can be made for any other specific menu implementation.

You can add any extra menus you need through a rewrite or a child them, and you can even keep boilerplate include files with the menus ( and other elements ) you commonly add in your design process.

If you can make the case that 80% or more of developers will use any of these other locations, we could consider adding them on a case-by-case basis, but I would bet that the use of all of these at once comes nowhere near 80%.

**Reply by PortMacOnline** [reply]
I agree that more than 80% of people will use that main menu location, but that wasn't my point -- my point was that we need more options than just ONE in 2021! I have given you suggestions, if you choose not to implement them that is your choice.

**Reply by 0dp** [reply]
In that case, i would merely suggest to follow the wordpress way of adding custom navigations and use the bootstrap nav walker.

This could be an entry in the docs for sure

**Reply by danemorgan** [reply]
Hi @PortMacOnline 

It isn't as simple as "my choice." If I made you feel your suggestion wasn't valued, I expressly apologize for doing that.

Thye point about the 80% is that as a rule, we don't want to add things to the main theme that fewer than 80% of developers are going to use. Those are things we expect developers to add on a case-by-case basis through rewriting the main theme or adding a child theme.

If there are other menu areas ( one of the footer areas seems likely, possibly ) that 80% of sites will use, then that would be a great candidate for a feature request. But we need to establish that there is that level of need out there first.

You seem passionate about this, and we would welcome your input on just how much any of these locations need to be supported by the core theme as opposed to leaving it to the developers. But we don't want to get into a situation where we add a menu location that 30 people are using and 70 people have to go in and delete every time they do a new build.

Does that make sense?

**Reply by danemorgan** [reply]
@0dp That's changing this year too. It's a hardline goal to roll out BS5 this year.

**Reply by 0dp** [reply]
> @0dp That's changing this year too. It's a hardline goal to roll out BS5 this year.

Cool, that discussion needs to be warmed up again

**Reply by PortMacOnline** [reply]
The reason I sounded "passionate" about it was because [a] multiple menus are a necessity for business websites in 2021, and [b] Brian Coords, who is a big fan of the theme, and teaches classes on using it, agreed that it would be a great addition to the theme and that I should fill in a feature request. So I went through the hassle of setting up an account here so that I could post the feature request, only to have it immediately hijacked and moved to a Discussion area!

Now the idea itself has been knocked on the head unless I can come up with enough evidence that it is needed. So I have done my part, do whatever you want with the theme, I will just go back to using one that already meets my needs. FWIW, attached is a screenshot from the free version of the Blocksy theme, which comes with **four** menu locations as standard, even in the free version. Apparently they meet WordPress's 80/20 rule.

![Screenshot from 2021-08-11 13-43-29](https://user-images.githubusercontent.com/7662890/128967822-61aa6e58-2532-476c-9572-3a86075ec709.png)

**Reply by 0dp** [reply]
I am happy to hear you are passionate about the cause and hope it wasn't too much of a hassle to create an account on GitHub.

It's not that your idea is not valid or not welcomed in the community. It's also not a bad thing that it's been moved to the discussion area. Just so we're clear on that 🙂

Now from my perspective, I use understrap to create custom solutions for clients who wish for a custom WordPress theme based on bootstrap.
It gives me a 'almost' blank canvas to work from where the most fundamental aspects have been configures. So that I can jump right in and start delivering results quick!

I don't know the theme you are referring to specifically, but i do know that a lot of themes gives this option; to choose different menu locations. I am not against it at all.

But for me personally, I'd rather want to add the menu locations i need myself, rather than having to remove them. While having several menu locations does not have a performance impact or anything. It's important for me that my clients get a solution that offers the agreed upon specs - no more no less. 

So in conclusion, to me, less is more.

As for a specific solution to your problem, as said before. Registering your own menu areas should not be too much of a hassle. There are many tutorials out there that can help if you are unsure how to proceed with it. There's also people here that would be keen to help. There are also some plugins that can help you, though i have never tried those myself.

Finally, i want to reiterate, the discussion is not shut down. We would just like for more community members to pitch in.

**Reply by bacoords** [reply]
Agree with @0dp here: We've recently been updating the process between 'issues' and 'discussions' so that issue are more about errors/bugs and discussions are places for everyone to chat about new features/enhancements. That's why we moved the discussion here, not as a slight (but I don't like how Github makes it feel like the original issue was 'closed/removed' because it's a little off-putting). 

I assume that the primary audience for Understrap is a developer who writes code, and thus wants to use Bootstrap-type elements in their theme building. That's the lens through which I see each decision, though I'm happy to hear differing opinions on that [on this thread](https://github.com/understrap/understrap/discussions/1291). @0dp makes a good point: when we deliver sites to a client, what features we turn OFF are sometimes just as important as what we turn ON, otherwise we get into a situation where a client turns something on that was out of scope and they want to know why it's not perfectly matching their design or something.

As for additional menus, with any feature, I think everyone just wants to see HOW and WHY we'd add it, so being more specific is helpful, like including the example of Blocksy. Otherwise, as others have said, you can modify this theme however you want using standard WordPress practices. So here's my thoughts:

My vote is for one additional menu in the footer. I think that having links to terms, privacy policy, etc at the bottom of your site is VERY common and passes the 80/20 rule. The hard-coded footer colophon has never seemed ideal to me, so I'd like to add a customizer setting to override the text (I'm working on this) and an optional nav menu that outputs a horizontal list after the colophon text.

Another option (based on the screenshots here) would be a separate mobile navigation menu. I don't personally see the value in that since Bootstrap handles the responsive menus very well. It would also be a pretty big structural shift, one that might be unnecessary with BS5 offering [Offcanvas](https://getbootstrap.com/docs/5.1/components/offcanvas/). My vote would be against adding that option for now, but returning to it with BS5. 

The other option shown above (one that's fairly common in starter themes) is a secondary header navigation, and my guess it that it's shown ABOVE the main site header. I've added to this to a number of Understrap child themes (I think we even built it as an example lesson in the [Understrap course](https://learn.understrap.com) ), but I'm on the fence about it as a whole. I'd want to hear from others.

One final note, while an idea may not make it into the core Understrap parent theme, it's still worth gauging interest, because someone might want to build it as an add-on or code example that others can use, so more discussion is key.

**Reply by PortMacOnline** [reply]
"I assume that the primary audience for Understrap is a developer who writes code, and thus wants to use Bootstrap-type elements in their theme building. "

I agree with that, and what you and 0dp are saying. I started building websites by writing static HTML using a text editor back in the late 90s/early2000s, so discovering Wordpress even in its early years was a revelation for me. Nowadays we build two types of business websites using WP, budget websites using pre-built templates and starter sites and customizing them, and premium websites using child themes of starter themes like GeneratePress and now Understrap and customizing them to suit.

But the **whole point** of starting with a starter theme is to not have to re-invent the wheel all the time, so if I need to hack the code of Understrap just to add basic functions, then it defeats the whole purpose of using it, I might as well start from scratch with something like Bootstrap5.

"I've added to this to a number of Understrap child themes (I think we even built it as an example lesson in the Understrap course"

That was how this whole discussion started, I asked you in the course discussion whether you had any code snippets to add to the child theme, and you suggested raising the issue here.

**Reply by 0dp** [reply]
Ohh boy @bacoords  🙈
